### ISSUE TITLE
Add a screen to onboarding for enabling notification permissions

### ISSUE THREAD

**Issue by GBKS** [BitcoinDesign/Bitcoin-Core-App#54]
Currently, when starting the app for the first time, the user is instantly greeted with a request to enable notifications. At this point, they are less likely to be able to make an informed decision since they have not gone through the first-use explainers, and the request modal does not come with a rationale. We can move this modal to a later point in the flow Here's a mock-up for a screen.

![image](https://user-images.githubusercontent.com/695901/236750651-14b73f01-f40f-41a5-a996-c0ab6f8f0cbb.png)

Tapping "Enable" would show the OS modal to enable the notification permission. Tapping "Skip" would simply go to the next screen.

A question I have is whether we then also need a new option in settings for this? Whether an app can send notifications or not is usually handled in the OS settings, not inside the app itself. But let's say a user enables this later, will the app know about this and can then create the system notification?

Feedback appreciated, as always.

**Comment by jarolrod** [comment]
cc @johnny9 

Can we implement this now? And can we then disconnect or reconnect the notification server through a settings menu option post-onboarding?

**Comment by johnny9** [comment]
Yes it is possible. It somewhat depends on https://github.com/bitcoin-core/gui-qml/pull/331 since the jni bridge will be used to implement it.

**Comment by GBKS** [comment]
This is now documented in the design docs [first use page](https://bitcoincore.app/first-use/) (very bottom).

**Comment by GBKS** [comment]
Going to close this design issue. Still needs to be implemented, but the design is done.
